build(windows): vendor xxd.c

benarcher2691 opened the pull request:
## Summary

- Vendor `xxd.c` from Vim, similar to how `tee.c` was added in #36363
- Build `xxd` executable as part of the Neovim build process
- Add `xxd.c` to lint and format targets

## Test plan

- [x] `xxd` builds successfully
- [x] `xxd --version` works
- [x] Round-trip test (`echo test | xxd | xxd -r`) works
- [x] Functional tests pass
- [x] Lint checks pass

Fixes #36664

benarcher2691 committed:
build(windows): vendor xxd.c

Problem:
Currently we fetch a prebuilt xxd.exe from neovim/deps for Windows,
which is not ideal in terms of obviousness, misses updates, and is
clumsy to update.

Solution:
Similar to tee.c (#36363), vendor xxd.c from Vim and build it as part
of the Neovim build process. This makes the source obvious, enables
analyzer checks, and simplifies updates.

Fixes #36664

justinmk commented on src/nvim/CMakeLists.txt:
```diff
@@ -427,6 +427,7 @@ list(SORT NVIM_HEADERS)
 
 list(APPEND UNCRUSTIFY_NVIM_SOURCES ${NVIM_SOURCES} ${NVIM_HEADERS})
 list(APPEND UNCRUSTIFY_NVIM_SOURCES ${PROJECT_SOURCE_DIR}/src/tee/tee.c)
+list(APPEND UNCRUSTIFY_NVIM_SOURCES ${PROJECT_SOURCE_DIR}/src/xxd/xxd.c)
```
nit (same for the other case below):

```
list(APPEND UNCRUSTIFY_NVIM_SOURCES
  ${PROJECT_SOURCE_DIR}/src/tee/tee.c
  ${PROJECT_SOURCE_DIR}/src/xxd/xxd.c
)
```

justinmk commented:
Thanks! Can you run `make formatc` and add those changes as a separate commit.

And one other small thing, let's add a trivial test that exercises `xxd` to confirm it basically works. It can live in `test/functional/editor/xxd_spec.lua`

benarcher2691 committed:
refactor(build): improve CMakeLists formatting, add xxd test

- Reformat list(APPEND ...) to multi-line style for readability
- Add functional test for xxd round-trip encoding/decoding

justinmk commented on build.zig:
```diff
@@ -365,6 +365,18 @@ pub fn build(b: *std.Build) !void {
     test_deps.dependOn(test_fixture(b, "streams-test", libuv, target, optimize, &flags));
 
+    // xxd - hex dump utility (vendored from Vim)
+    const xxd_exe = b.addExecutable(.{
```
oh glad you noticed this. i think we forgot to update build.zig for `tee` ...

(not a blocker for this PR)

justinmk reviewed (approved):
Perfect, thanks!
